src/db.c

```diff
@@ -1739,8 +1740,20 @@ int getKeysUsingKeySpecs(struct redisCommand *cmd, robj **argv, int argc, getKey
         int count = ((last - first)+1);
         keys = getKeysPrepareResult(result, count);
 
+        /* Not enough keys were provided to meet the spec, which is probably
+         * a user error, so we will try to provide the keys by clamping the last
+         * value. Note it's possible for an overflow to have occurred, in which
+         * case last will be less than first. */
+        if (last > argc || last < first) last = argc;
```

Comment:
off by one? (last is an actual position, and argc is a count)
```suggestion
        if (last >= argc || last < first) last = argc-1;
```